warehouse/query-core/src/main/java/datawave/query/transformer/ExcerptTransform.java

```diff
@@ -84,12 +91,111 @@ public Entry<Key,Document> apply(@Nullable Entry<Key,Document> entry) {
+     *            The phrase indexes in map form to speed up searches.
+     * @return The TermWeightPosition for the given hit term
+     */
+    private TermWeightPosition getOffset(ValueTuple hitTuple, Map<String,Map<String,PhraseIndexes>> phraseIndexMap) {
+        Key docKey = hitTuple.getSource().getMetadata();
+        // if we do not know the source document key, then we cannot find the term offset
+        if (docKey == null) {
+            log.warn("Unable to find the source document for a hit term, skipping excerpt generation");
             return null;
         }
+        String fieldName = hitTuple.getFieldName();
+        String eventId = keyToEventId(docKey);
+        
+        // get the key at which we would find the term frequencies
+        Key tfKey = new Key(docKey.getRow().toString(), Constants.TERM_FREQUENCY_COLUMN_FAMILY.toString(), docKey.getColumnFamily().toString() + '\u0000'
+                        + hitTuple.getValue() + '\u0000' + hitTuple.getFieldName());
+        Range range = new Range(tfKey, tfKey.followingKey(PartialKey.ROW_COLFAM_COLQUAL));
+        try {
+            // seek directly to that key
+            source.seek(range, Collections.emptyList(), false);
+            if (source.hasTop()) {
+                // parse the term frequencies
+                TermWeight.Info twInfo = TermWeight.Info.parseFrom(source.getTopValue().get());
+                TermWeightPosition.Builder position = new TermWeightPosition.Builder();
+                
+                // if we have phrase indexes, then find one that overlaps if any
+                // pull out the appropriate phrase indexes for this field/event id
+                if (phraseIndexMap.containsKey(hitTuple.getFieldName())) {
+                    PhraseIndexes phraseIndexes = phraseIndexMap.get(fieldName).get(eventId);
+                    if (phraseIndexes != null) {
+                        for (int i = 0; i < twInfo.getTermOffsetCount(); i++) {
+                            position.setTermWeightOffsetInfo(twInfo, i);
+                            TermWeightPosition pos = position.build();
+                            if (phraseIndexes.getOverlap(fieldName, eventId, pos.getLowOffset(), pos.getOffset()) != null) {
+                                // if we have an overlapping phrase, then return this position
+                                return pos;
+                            }
+                        }
+                    }
+                }
+                
+                // if no overlapping phrases, then return the first position
+                position.setTermWeightOffsetInfo(twInfo, 0);
+                return position.build();
+            }
+            
+        } catch (InvalidProtocolBufferException e) {
+            log.error("Value passed to aggregator was not of type TermWeight.Info for " + tfKey, e);
+        } catch (IOException e) {
+            log.error("Failed to scan for term frequencies at " + tfKey, e);
+        }
```

Comment:
Consider whether we want to swallow and log these -or- whether we should throw them updwards and fail the query? It seems that they might be sufficiently bad to fail upwards, but also, other things would probably fail if either of these happened, so the extra complexity might not be necessary.